src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -706,6 +709,67 @@ public static IResourceBuilder<T> WithHealthCheck<T>(this IResourceBuilder<T> bu
         return builder;
     }
 
+    /// <summary>
+    /// Adds a health check to the resource which is mapped to a specific endpoint.
+    /// </summary>
+    /// <typeparam name="T">A resource type that implements <see cref="IResourceWithEndpoints" />.</typeparam>
+    /// <param name="builder">A resource builder.</param>
+    /// <param name="path">The relative path to test.</param>
+    /// <param name="statusCode">The result code to interpret as healthy.</param>
+    /// <param name="endpointName">The name of the endpoint to derive the base address from.</param>
+    /// <returns>A resource builder.</returns>
+    public static IResourceBuilder<T> WithHttpHealthCheck<T>(this IResourceBuilder<T> builder, string? path = null, int? statusCode = null, string? endpointName = null) where T : IResourceWithEndpoints
```

Comment:
This overloading feels a little strange, i.e. `WithHttpHealthCheck` allowing specifying an endpoint name that isn't "http", and vice-versa for `WithHttpsHealthCheck`. Should we consider having a `WithEndpointHealthCheck` that these two methods just call down into passing the appropriate endpoint name instead?

Reply:
What protocol would be using for WithEndpointHealthCheck. I know HTTP rules the world but ....

Reply:
It would obtain it from the endpoint specified and if it's not an HTTP-based scheme it would throw.

Reply:
This should be coupled to http, not endpoints in general. Throw if the scheme isn't http as well.

Reply:
So should `WithHttpsHealthCheck` throw if the scheme of the endpoint specified isn't `https`? Or is it just a shortcut for specifying the endpoint named "https" and it throws if the scheme of that endpoint isn't "http" or "https"?

Reply:
Yes. I'm doing that now and have covering tests.